src/analysis/processing/qgsalgorithmpointstolines.cpp

```diff
@@ -0,0 +1,226 @@
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpointstolines.h"
+#include "qgsvectorlayer.h"
+
+///@cond PRIVATE
+
+QString QgsPointsToLinesAlgorithm::name() const
+{
+  return QStringLiteral( "pointstolines" );
+}
+
+QString QgsPointsToLinesAlgorithm::displayName() const
+{
+  return QObject::tr( "Points to lines" );
+}
+
+QString QgsPointsToLinesAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm takes a point layer and connects its features creating a new line layer.\n\n"
+                      "An attribute or expression may be specified to define the order the points should be connected. "
+                      "If no order expression is specified, the fid is used.\n\n"
+                      "A natural sort can be used when sorting by a string attribute "
+                      "or expression (ie. place 'a9' before 'a10').\n\n"
+                      "An attribute can be selected to group points having the same value into the same resulting line." );
+}
+
+QStringList QgsPointsToLinesAlgorithm::tags() const
+{
+  return QObject::tr( "create,lines,points,connect,convert,join" ).split( ',' );
+}
+
+QString QgsPointsToLinesAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsPointsToLinesAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+void QgsPointsToLinesAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "CLOSE_LINES" ),
+                QObject::tr( "Create closed lines" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "ORDER_EXPRESSION" ),
+                QObject::tr( "Order expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "NATURAL_SORT" ),
+                QObject::tr( "Sort text containing numbers naturally" ), false, true ) );
+  addParameter( new QgsProcessingParameterField( QStringLiteral( "GROUP_FIELD" ),
+                QObject::tr( "Line group field" ), QVariant(), QStringLiteral( "INPUT" ), QgsProcessingParameterField::Any, false, true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ),
```

Comment:
Also needs DATE_FORMAT parameter ported

Reply:
@nyalldawson I am afraid this will need a complete `strptime` implementation, just for compatibility sake of a poorly documented parameter. Got any suggestion?

Reply:
We could potentially make a converter from strptime format strings to the qt QDateTime::fromString formats, (see https://doc.qt.io/qt-5/qtime.html#fromString-1), at least for the common format parameters.

Reply:
We could use `strptime` for linux builds and a map from strptime to qdatetime strings for windows, but that will not be very pretty and probably not fully backwards compatible! I'd really rather maintain the python code, but it's your call!